addons/website/static/src/scss/website.scss

```diff
@@ -246,7 +246,19 @@ body:not(.o_fullscreen) {
     @include o-apply-colors('menu-custom');
     $-menu-color: o-color('menu-custom') or o-color('menu');
     @if ($-menu-color and color-yiq($-menu-color) != $yiq-text-dark) {
-        @extend .navbar-dark;
+        // ":not(.bg-light)" was added as a fix in the stable version because,
+        // without it, the two classes ".navbar-light" and ".bg-light" did not
+        // work together correctly. For instance, if a ".navbar-light" element
+        // had a dark background (resulting in light text), combining it with
+        // the "bg-light" class would keep the background light, rendering the
+        // light text invisible.
+        // TODO: In the Master, we need to ensure that these two classes
+        // complement each other by applying a dark background to "bg-light"
+        // elements when necessary, similar to the approach taken with
+        // "navbar-light."
+        &:not(.bg-light) {
+            @extend .navbar-dark;
+        }
```

Comment:
@qsm-odoo It feels fine for me, but I'd love to have your opinion on this workaround.

Reply:
> It feels fine for me, but I'd love to have your opinion on this workaround.

I think that's what I already discussed with BVR IRL so I think it is fine. However:

> by applying a dark background to "bg-light" elements when necessary

I don't think that will be the plan: we'll rather need to prevent adding the bg-light class not? Or just change our navbar-light rule here into something like navbar-default-colors or something like that ? :shrug: 

--

I don't think the "not bg-light" condition should be around the `extend` only? I see there is a `@include o-add-gradient('menu-gradient');` below, does not that mean that if there is a gradient, it will override the bg-light since this is adding a background image? and making this diff would thus broke that case? Maybe the "not bg light condition" should be on the whole `.navbar-light` rule?

Also about the added specificity due to the use of `:not`, I think there is actually a solution for stable: using `:where(:not(.bg-light))` which should do the same but without increasing the specificity. In master, to avoid though.

Reply:
> I don't think that will be the plan: we'll rather need to prevent adding the bg-light class not? Or just change our navbar-light rule here into something like navbar-default-colors or something like that ? 🤷

Maybe removing 'bg-light' would be the most appropriate. Look:

![image](https://github.com/odoo/odoo/assets/52911687/32f3a632-d880-4b2f-bc49-3e315c7bae92)

Reply:
> I don't think the "not bg-light" condition should be around the extend only? I see there is a @include o-add-gradient('menu-gradient'); below, does not that mean that if there is a gradient, it will override the bg-light since this is adding a background image? and making this diff would thus broke that case? Maybe the "not bg light condition" should be on the whole .navbar-light rule?

> Also about the added specificity due to the use of :not, I think there is actually a solution for stable: using :where(:not(.bg-light)) which should do the same but without increasing the specificity. In master, to avoid though.

Indeed, I didn't know this `:where` .. This is exactly what I need here to put the `:not` around all the rules

Reply:
> I don't think that will be the plan: we'll rather need to prevent adding the bg-light class not? Or just change our navbar-light rule here into something like navbar-default-colors or something like that ? 🤷

To be tested in Master, I will modify the comment/commit